app/javascript/components/notification-drawer/notification-drawer.jsx

```diff
@@ -0,0 +1,192 @@
+import React, { useState } from 'react';
+import moment from 'moment';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Button, NotificationDrawer as Drawer, Notification, MenuItem, NotificationContent, NotificationInfo, NotificationMessage, Icon,
+} from 'patternfly-react';
+import classnames from 'classnames';
+import { getNotficationStatusIconName, unreadCountText, viewDetails } from './helpers';
+import { maxNotifications as maxNotificationsConstant } from '../../packs/notification-drawer-common';
+import {
+  toggleDrawerVisibility, markNotificationRead, markAllRead, clearNotification, clearAll, toggleMaxNotifications,
+} from '../../miq-redux/actions/notifications-actions';
+
+const NotificationDrawer = () => {
+  const dispatch = useDispatch();
+  const drawerTitle = __('Notifications');
+  const [isDrawerExpanded, setDrawerExpanded] = useState(false);
+  const [isPanelExpanded, setPanelExpanded] = useState(true);
+  const {
+    isDrawerVisible, unreadCount, notifications, totalNotificationsCount, maxNotifications,
+  } = useSelector(({
+    notificationReducer: {
```

Comment:
There is a lot of extra code duplications. If you have a use case like this, when you are basically consuming the whole state, you can just return the whole object and destruct it only in one place:

```jsx
const {
    isDrawerVisible, unreadCount, notifications, totalNotificationsCount, maxNotifications,
  } = useSelector(({ notificationReducer }) => notificationReducer)
```